### ISSUE TITLE
.netrc not honored if auth-type is used

### ISSUE THREAD

**Issue by pszlazak** [jakubroztocil/httpie#852]
Once `--auth-type` switch is used, `.netrc` is not honored. Authentication details must be provided via `--auth` switch.

Details:
```
# http --debug --auth-type=basic example.org
HTTPie 0.9.8
Requests 2.19.1
Pygments 2.2.0
Python 2.7.13 (default, Sep 26 2018, 18:42:22)
[GCC 6.3.0 20170516]
/usr/bin/python
Linux 4.9.0-9-amd64

<Environment {
    "colors": 8,
    "config": {
        "__meta__": {
            "about": "u'HTTPie configuration file'",
            "help": "u'https://httpie.org/docs#config'",
            "httpie": "u'0.9.8'"
        },
        "default_options": "[]"
    },
    "config_dir": "/root/.httpie",
    "is_windows": false,
    "stderr": "<open file '<stderr>', mode 'w' at 0x7fd6490091e0>",
    "stderr_isatty": true,
    "stdin": "<open file '<stdin>', mode 'r' at 0x7fd6490090c0>",
    "stdin_encoding": "UTF-8",
    "stdin_isatty": true,
    "stdout": "<open file '<stdout>', mode 'w' at 0x7fd649009150>",
    "stdout_encoding": "UTF-8",
    "stdout_isatty": true
}>
usage: http [--json] [--form] [--pretty {all,colors,format,none}]
            [--style STYLE] [--print WHAT] [--headers] [--body] [--verbose]
            [--all] [--history-print WHAT] [--stream] [--output FILE]
            [--download] [--continue]
            [--session SESSION_NAME_OR_PATH | --session-read-only SESSION_NAME_OR_PATH]
            [--auth USER[:PASS]] [--auth-type {basic,digest}]
            [--proxy PROTOCOL:PROXY_URL] [--follow]
            [--max-redirects MAX_REDIRECTS] [--timeout SECONDS]
            [--check-status] [--verify VERIFY]
            [--ssl {ssl2.3,tls1,tls1.1,tls1.2}] [--cert CERT]
            [--cert-key CERT_KEY] [--ignore-stdin] [--help] [--version]
            [--traceback] [--default-scheme DEFAULT_SCHEME] [--debug]
            [METHOD] URL [REQUEST_ITEM [REQUEST_ITEM ...]]
http: error: --auth required
```

**Comment by nb2838** [comment]
I'd like to tackle this if that's ok!

**Comment by jakubroztocil** [comment]
@nb2838 Sure, but maybe after or instead of #926 (which you also expressed interest in)? So things don’t get blocked.  Btw, do you happen to be from @MLH-Fellowship?

**Comment by nb2838** [comment]
Hey! No, I am not a part of MLH-Fellowship. I am working with a group of friends on tackling small projects. I asked for both so that my friend and I could work on these two issues together. I am sorry, I should have been more clear about it. Would it be ok for us to work on them?

**Comment by jakubroztocil** [comment]
Sounds good 👍

**Comment by nb2838** [comment]
I've been taking a look at the bug and part of the issue lies in how the program deals with the .netrc file. Currently, if I am not mistaken, it is done by letting the library "requests" deal with it. The problem is that according to  requests documentation it seems that they only support reading the .netrc file when no authentication parameters are specified and only for Basic Authentication 
![image](https://user-images.githubusercontent.com/61052993/84420397-97bca200-abdf-11ea-87a4-71eec1b95b35.png)
I was wondering then if it would be better to deal with this in httpie itself by implementing this functionality maybe as a utility rather than dealing with it through requests. This way we could implement support for using Digest Authentication with .netrc which currently requests doesn't support.

**Comment by nb2838** [comment]
I think I have to change the way the program deals with the reading of the .netrc file. Currently, it does by delegating the task to the requests library. However, the way in which this is done doesn't allow for Digest authentication which is why we would need to do it on our own and change that. When I was doing this I came with an issue/doubt. For security reasons, the python function in the standard library doesn't allow to read the .netrc  file if the permissions are not strict enough for security reasons. I was wondering if httpie should do the same thing and if I should implement this functionality.

**Comment by jakubroztocil** [comment]
This is now addressed in current `master` via b865988. Feedback would be appreciated.
